nixos/modules/services/databases/postgresql.md

```diff
@@ -170,6 +170,41 @@ are already created.
   }
 ```
 
+## Authentication {#module-services-postgres-authentication}
+
+Local connections are made through unix sockets by default and support [peer authentication](https://www.postgresql.org/docs/current/auth-peer.html).
+This allows system users to login with database roles of the same name.
+For example, the `postgres` system user is allowed to login with the database role `postgres`.
+
+System users and database roles might not always match.
+In this case, to allow access for a service, you have two options:
+
+  1. Create a database role matching your system user during [initialization](#module-services-postgres-initializing).
```

Comment:
I think it's a little odd to suggest creating a role matching yoru system user if the precondition is that those don't match ;)